src/Symfony/Component/HttpFoundation/File/MimeType/MimeTypeGuesser.php

```diff
@@ -125,18 +117,40 @@ public function guess($path)
             throw new AccessDeniedException($path);
         }
 
-        if (!$this->guessers) {
+        if (!$guessers = $this->getGuessers()) {
             $msg = 'Unable to guess the mime type as no guessers are available';
             if (!FileinfoMimeTypeGuesser::isSupported()) {
                 $msg .= ' (Did you enable the php_fileinfo extension?)';
             }
             throw new \LogicException($msg);
         }
 
-        foreach ($this->guessers as $guesser) {
+        foreach ($guessers as $guesser) {
             if (null !== $mimeType = $guesser->guess($path)) {
                 return $mimeType;
             }
         }
     }
+
+    /**
+     * @return MimeTypeGuesserInterface[]
+     */
+    private function getGuessers()
+    {
+        if (!$this->nativeGuessersLoaded) {
```

Comment:
This is not going to work in case someone registers new guessers later with `register()`. Then native guessers will never be appended and that might affect the guessing process since there will be no default guessers.